keras_rcnn/metrics.py

```diff
@@ -0,0 +1,4 @@
+import tensorflow
+
+def mean_average_precision(y_true, y_pred):
+    return tensorflow.reduce_mean(tensorflow.metrics.sparse_average_precision_at_k(tensorflow.cast(y_true, tensorflow.int64), y_pred, 1)[0])
```

Comment:
We need to implement a custom average precision function because tensorflow.metrics.sparse_average_precision_at_k expects to be updated in training (i.e. it’s incompatible with Keras’s per-batch metrics), e.g.

```Python
keras.backend.mean(keras.backend.cumsum(a) / keras.backend.cumsum(b))
```